src/tracker/inflights.rs

```diff
@@ -122,6 +127,23 @@ impl Inflights {
         self.count = 0;
         self.start = 0;
     }
+
+    /// free buffer memory by reassignment
+    #[inline]
+    pub fn free_mem(&mut self) {
+        let mut v = Vec::with_capacity(self.count);
+        for i in 0..self.count {
+            v.push(self.buffer[self.start + i])
```

Comment:
I don't get it. `v` doesn't seem to be used, and why append the element from buffer to `v`? On the other hand, `self.buffer` should be dropped to free memory.

Reply:
I am trying to re-create an `buffer` with `length` & `capacity` equal to `count`, in order to decrease the allocated memory, 
and remain only from `self.start` to `self.start + self.count - 1`

> On the other hand, `self.buffer` should be dropped to free memory

If  `self.count` is zero, `self.buffer` will be `Vec` with capacity 0, I am not sure what is the implementation inside, but I guess there is no memory allocated?
If  `self.count` is positive, we should at least remain `self.count` sized Vec (?

Reply:
`shrink` that mentioned before, I think it will do nothing if the `Vec` is full once

Reply:
What is the issue left related to the memory problem? Is #10156?

Reply:
What is 10156?

Reply:
Sorry, is this https://github.com/tikv/tikv/issues/10156, found it under https://github.com/tikv/rfcs/pull/59

Reply:
/cc @hicqu

Reply:
The left issue is unchanged for raft-rs. Memory usage of `Inflight` is too large if there are lots of raft peers.

Reply:
@BusyJay do you mean `free_mem` should only free empty inflights, but keep no-empty inflights unchanged?

Reply:
I have discussed with Jay. He thinks the current implemenatation is OK. BTW how about using `vec.extend_from_slice` instead?